### PR TITLE
[ECP-9037] - Support Dispute webhooks on webhook module

### PR DESCRIPTION
**Pull request by raoulritter**
## Tested scenarios

Added webhook module support 

**Fixed issue**:

### CODE REVIEW

File: src/Processor/AbstractDisputeNotificationProcessor.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ */
+
+namespace Adyen\Webhook\Processor;
+
+use Adyen\Webhook\PaymentStates;
+
+
```
Comment (candemiralp) [inline src/Processor/AbstractDisputeNotificationProcessor.php:28]:
```suggestion
```
Formatting

File: src/Processor/AbstractDisputeNotificationProcessor.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    const DISPUTE_STATUS_ACCEPTED = "Accepted";
+    const FINAL_DISPUTE_STATUSES = [
+        self::DISPUTE_STATUS_LOST,
+        self::DISPUTE_STATUS_ACCEPTED,
+        self::DISPUTE_STATUS_UNDEFENDED
+    ];
+    const DISPUTE_STATUS = "disputeStatus";
+
+
+    const ORDER_ARRAY = [
```
Comment (candemiralp) [inline src/Processor/AbstractDisputeNotificationProcessor.php:43]: Can you please rename it to more self-explanatory name? `CHARGEBACK_ORDER_STATES` maybe

File: src/Processor/ProcessorFactory.php [hunk 3]
```diff
@@ -51,7 +51,9 @@ class ProcessorFactory
         EventCodes::VOID_PENDING_REFUND => VoidPendingRefundProcessor::class,
         EventCodes::CHARGEBACK => ChargebackProcessor::class,
         EventCodes::CHARGEBACK_REVERSED => ChargebackReversedProcessor::class,
```
Comment (candemiralp) [inline src/Processor/ProcessorFactory.php:53]: I couldn't see the updated processor for `CHARGEBACK_REVERSED`. Don't we need to modify it?

File: src/Processor/AbstractDisputeNotificationProcessor.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        $state = $this->initialState;
+        $additionalData = $this->notification->getAdditionalData();
+        $disputeStatus = $additionalData[self::DISPUTE_STATUS] ?? null;
+
+        if ($this->notification->isSuccess() &&
+            isset($disputeStatus) &&
+            in_array($disputeStatus, self::FINAL_DISPUTE_STATUSES)) {
+            //State should be In Progress or paid in cases where we want to change it to refunded.
+            if (in_array($state, self::ORDER_ARRAY)) {
```
Comment (candemiralp) [inline src/Processor/AbstractDisputeNotificationProcessor.php:58]: You can remove this `if` block and add the condition to the previous block.

File: src/Processor/ProcessorFactory.php (hunk 3 above)
Comment (raoulritter) [reply src/Processor/ProcessorFactory.php:53]: No we don't have to modify the chargeback_reversed processor as this was already present

File: src/Processor/AbstractDisputeNotificationProcessor.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    const DISPUTE_STATUS_LOST = "Lost";
+    const DISPUTE_STATUS_ACCEPTED = "Accepted";
+    const FINAL_DISPUTE_STATUSES = [
+        self::DISPUTE_STATUS_LOST,
+        self::DISPUTE_STATUS_ACCEPTED,
+        self::DISPUTE_STATUS_UNDEFENDED
+    ];
+    const DISPUTE_STATUS = "disputeStatus";
+
+
```
Comment (candemiralp) [inline src/Processor/AbstractDisputeNotificationProcessor.php:40]:
```suggestion
```

**Review by candemiralp — APPROVED** [review]

**Review by peterojo — APPROVED** [review]

**Pull request merged**